### PR TITLE
Send identify events to Segment before page events from ecommerce

### PR DESCRIPTION
**Pull request by mikedikan**
ECOM-6650

Calling the logging of the user to segment before calling the tracking function, so events are attributed to the correct user.  Also, small cleanup of settings file.

@edx/ecommerce for review

### CODE REVIEW

File: ecommerce/static/js/views/analytics_view.js [hunk 1]
```diff
@@ -65,20 +65,19 @@ define([
...
-             * Get data for initializing segment.
+             * Get data for initializing segment and records the initial page load.
              */
-            getSegmentPageData: function () {
+            recordPageData: function () {
+                var pageData = {};
+
                 if (this.options.courseModel.get('courseId')) {
-                    return this.buildCourseProperties();
+                    pageData= this.buildCourseProperties();
```
Comment (AlasdairSwan) [inline ecommerce/static/js/views/analytics_view.js]: Nitpick: missing space between ```pageData``` and ```=```.

**Review by AlasdairSwan — CHANGES_REQUESTED** [review]
Just 1 nit.

File: ecommerce/settings/base.py [hunk 2]
```diff
@@ -426,13 +426,6 @@
 # END AUTHENTICATION
 
 
-# ANALYTICS
-# Specify a key to emit events to the corresponding Segment project. `None` disables tracking.
-# See: https://segment.com/docs/libraries/python/
-SEGMENT_KEY = None
```
Comment (rlucioni) [inline ecommerce/settings/base.py:435]: If this is unused, it should also be removed from [edx/configuration](https://github.com/edx/configuration/blob/master/playbooks/roles/ecommerce/defaults/main.yml#L70-L71); you may want to consult with Devops to see if any secure overrides need to be cleaned up.

Related: it's probably worth documenting how this is meant to be set up now.

**Review by rlucioni — APPROVED** [review]

File: ecommerce/settings/base.py (hunk 2 above)
Comment (mikedikan) [reply ecommerce/settings/base.py:435]: Good call, thanks @renzo for the suggestion

File: ecommerce/static/js/views/analytics_view.js (hunk 1 above)
Comment (mikedikan) [reply ecommerce/static/js/views/analytics_view.js]: nit accepted

File: ecommerce/settings/base.py (hunk 2 above)
Comment (mikedikan) [reply ecommerce/settings/base.py:435]: @rlucioni https://openedx.atlassian.net/browse/DEVOPS-5416

**Comment by mikedikan** [comment]
@AlasdairSwan please re-review

**Review by AlasdairSwan — APPROVED** [review]
:shipit:

**Pull request merged**